### DISCUSSION TITLE
Make UI control for tags view/editing

### DISCUSSION THREAD

**Discussion by SwiftExtender** [AvaloniaUI/Avalonia#15752] [Q&A]
I want to create control, which let me to input many values with autocomplete for each value. This behavior seems perfect for tags management. I see something like this in DevExpress (video related), but i cant figure out how to implement this in Avalonia. Any ideas how to make this? I think that i must implement textBox with embedded AutocompleteBoxes, but maybe i miss something?

https://github.com/AvaloniaUI/Avalonia/assets/69252550/fb944e80-c47a-4b1d-9aa9-d0e98402c5dd

**Comment by thevortexcloud** [comment]
I would do it by just using a standard textbox/autocomplete box and some internal HashSet containing all the unique tags. If the list validates you can put an underline on all the text. Which looks like it should match what that video does. If you however need to say, highlight when the caret is over a specific tag it gets a bit more complicated.

**Reply by SwiftExtender** [reply]
This anwser cant help to trigger autocomplete dropdown for every new word as seen in example(
Also there is no info about how to override type of Text property of AutoCompleteBox, because string property doesn`t fit for tag management
